packages/patterns/src/components/partial-syncing-controls.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { nanoid } from 'nanoid';
+
+/**
+ * WordPress dependencies
+ */
+import { BaseControl, CheckboxControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { InspectorControls } from '@wordpress/block-editor';
+
+function PartialSyncingControls( { attributes, setAttributes } ) {
+	// Only the `content` attribute of the paragraph block is currently supported.
+	const attributeName = 'content';
+
+	const isPartiallySynced =
+		attributes.connections?.attributes?.[ attributeName ]?.source ===
+		'pattern_attributes';
+
+	function updateConnections( isChecked ) {
+		if ( ! isChecked ) {
+			setAttributes( {
+				connections: undefined,
+			} );
+			return;
+		}
+		if ( typeof attributes.metadata?.id === 'string' ) {
+			setAttributes( {
+				connections: {
+					attributes: {
+						[ attributeName ]: {
+							source: 'pattern_attributes',
+						},
+					},
+				},
+			} );
+			return;
+		}
+
+		const id = nanoid( 6 );
+		setAttributes( {
+			connections: {
+				attributes: {
+					[ attributeName ]: { source: 'pattern_attributes' },
+				},
+			},
+			metadata: {
+				...attributes.metadata,
+				id,
```

Comment:
It might not be a final name while we're in the early stages of development, but I think the `id` could still be named a bit more descriptively.
- `connectionId` is an option, though I don't think it will be used by other types of connections (custom fields), so that could be misleading.
- `partialSyncId` might be better for now. Anyone that encounters it will know exactly what it relates to.
- `dynamicContentId` or `patternAttributeId` could also be options.

Reply:
I think `id` could potentially be good if we want to extend it to similar features in the future. Otherwise, we might also want to consider putting this id inside `connections` rather than `metadata`.

Reply:
Moving it to the connections object might be a good short-term measure. 

Metadata is quite general, so folks might consider it as used for an html id or something.

Reply:
Yeh, I can't remember the reason for using the new `metadata` attrib in that original PR. I think given where we are now at with the pattern syncing it will be safer to move it into the connections object so the purpose is clearer.

Reply:
I believe it's because the Block Naming API is using `metadata.name`? I don't mind switching it to `connections.id` for now but we can't really change it (well we can but it'll be painful 😆) after it's saved. AFAIK, the `metadata` attribute is a preserved attribute that only core could use it (or else the block naming API would be broken 😅). I'd prefer putting it inside `metadata` so that other future APIs could reuse it if they want something like a `serverId` or `internalId` (as opposed to `clientId`). Now that we're using `nanoid`, it should be pretty safe to assume that it's unique across blocks too. But yet again, we could always change back to `metadata.id` once we finalize the API in `connections` 😆.

Reply:
one option would be `metadata.staticId` to indicate it is an id that is permanent and serialised with content as distinct from the dynamic `clientId`